index.html

```diff
@@ -1,135 +1,169 @@
+         </ol>
+             
+      <p>So far the decoding mechanism does not deal gracefully with bad
+         passwords.  It should be possible to change only the first and
+         last character to fix the checksums.  But for some reason, when
+         you enter a bad password, it changes more than that.  I've also
+         got a few strange javascript errors I'm not sure how to deal with.
+         Regardless, it seems to work fine in Firefox 1.5.
+      <p>I've written a utility to enter passwords into an emulator
+         window automatically (in Linux at least), and may post it at some
+         point.
+      <p>The inspiration to crack this game, as well as how to write such
+         a javascript application are due primarily to Bisqwit 
+         (http://iki.fi/bisqwit/) and I am therefore grateful for his work.
+      <p><a href="http://pages.physics.cornell.edu/~shicks/bolympus.html">Steve Hicks</a> (<a href="mailto:sdh33@cornell.edu">sdh33 at cornell edu</a>)
+      <h2>Coding Scheme</h2>
+      <p>BoO passwords have 26 characters from an alphabet of 64.  In order,
+         0-9 correspond to 0x00-0x09, A-Z to 0x0A-0x24, a-z to 0x25-0x3D, and
+         ? and ! to 0x3E and 0x3F.  Thus, each character is 6 bits, and all arithmetic
+         we will do from now on is thus modulo 64.
+      <p>The actual information content of the password is 23 characters long.
+         Two characters are used for checksums and one is arbitrary.  Call the
+         23 characters a(1) through a(23).  The first checksum is to pick a(0) such that
+         a(0)+...+a(23)=0x00.  Now pick an arbitrary character c(0).  Compute c(i+1)=[c(i)^a(i)]+d(i)
+         where d(i)=0x38,0x0A,0x12,0x26,... for i=0,1,2,3 and repeating with a period of
+         4 beyond that.
+      <p>We now have c(0)...c(24), giving 25 characters.  The last character c(25) is
+         computed soo that c(0)+...+c(25)=0x00.  If both of these checksums are satisfied,
+         then the password is valid.
+      <p>Regarding the value of each bit, use the following chart (in case the javascript doesn't show up):
+      <ul>
+         <li>0-3: location
+         <li>4-7: olives in BCD, ones place
+         <li>8-11: olives in BCD, tens place
+         <li>12-16: salamander skins
+         <li>17-52: hero name, 0x3F for blank
+         <li>53-88: heroine name
+         <li>89-94: life
+         <li>96: Crystal
+         <li>97: Flask
+         <li>98: Ocarina
+         <li>99: Harp
+         <li>100: Divine Sword
+         <li>101: Nymph Sword
+         <li>102: Staff of Fennel
+         <li>103: Club
+         <li>104: 2nd Heart
+         <li>105: 1st Heart
+         <li>106: Key
+         <li>107: Power Bracelet
+         <li>108: Fire Shield
+         <li>109: Reflective Shield
+         <li>110: Sandals
+         <li>111: Moonbeam
+         <li>112: Argolis Ambrosia
+         <li>113: Phthia Ambrosia
```

Comment:
Is this no longer accurate?

Reply:
I only checked the Phrygia Ambrosia since you had a ? next to it. Turned out that bit 116 didn't control that one, though I would have programmed it to do it that way. And the shields were switched. I didn't really check the rest.

I'm kind of curious to see the disassembly. I assume you used Ghidra or something?

As for the decoded password and your guess, that was exactly my guess as well. Honestly, easiest thing to do is find where those flags get copied to and just watch it for changes when you do things. Since most of the obvious ones are dealt with already, it wouldn't be hard to figure out when something else is changing. For all we know they could just be transistory flags that govern in-game behavior, but based on what I've seen when setting some of the unknowns, they just stay set until _who knows what changes them_.
And don't feel too bad about your old code. I look back at code I wrote when I was in undergrad and high school (and grad school) and I still think "Wow, who coded that mess. Oh, me."

Reply:
Sorry, I could have sworn I responded earlier.

I also think I may be mistaken about the disassembly - I'm not finding it, nor did I actually play through the game to generate it.  I use my own custom disassembling emulator, jsnesx.  You play through the game and it keeps track of code vs data and various indirection, etc, to generate a very usable dump.  There are some things that static analysis can do better, but generally this is easier and more reliable.  I've been meaning to play through the whole game again so that I can build a randomizer for it (I think still nobody has), and I'd probably do that in the disassembler so as to get a better dump.

Reply:
jsnesx doesn't look like you've posted it, but it might be fun to play with.

Reply:
https://github.com/shicks/jsnesx/blob/master/disassembler.md

Reply:
The UX is pretty poor, but once you get past that it's pretty usable.

Reply:
lol as long as I can make it run on linux, I'll be happy with a command line interface.